src/core/packet_builder.c

```diff
@@ -379,6 +379,27 @@ QuicPacketBuilderPrepare(
             Builder->Metadata->PacketId,
             Builder->BatchId);
 
+        //
+        // Occassionally skip a packet number for improved security.
+        //
+        if (Connection->Send.NextSkippedPacketNumber == Connection->Send.NextPacketNumber) {
+            Connection->Send.SkippedPacketNumber =
+                Connection->Send.NextPacketNumber++;
+            QuicTraceLogConnWarning(
+                SkipPacketNumber,
+                Connection,
+                "Skipped packet number %llu",
+                Connection->Send.SkippedPacketNumber);
+
+            //
+            // Randomly skip a packet number (from 0 to 65535).
```

Comment:
"The server skips a packet number with a uniform probability of at most one every 131,072 packets"

Just pasting the how quiche fixed the issue here just in case you missed it in the paper. I don't have an opinion on this.

Reply:
I decided to go a slightly different direction than quiche. 😄

Reply:
What are the chances that we send `NextSkippedPacketNumber` before we receive the hypothetical ack for `SkippedPacketNumber`? If we send skipped ack to fast and keep track only of the last one, we could defeat our own protection.

How fast are we generally consuming packets numbers? I don't have a good estimate in mind.

Reply:
It depends on the speed and RTT. But at the end of the day, this is a best effort feature.